pkg/build/builder/daemonless.go

```diff
@@ -306,10 +306,76 @@ func buildDaemonlessImage(sc types.SystemContext, store storage.Store, isolation
 func generateTransientMounts() []string {
 	mounts := []string{}
 	mounts = appendRHSMMount(mounts)
+	mounts = appendETCPKIMount(mounts)
+	mounts = appendRHRepoMount(mounts)
 	mounts = appendCATrustMount(mounts)
 	return mounts
 }
 
+func appendRHRepoMount(mounts []string) []string {
+	st, err := os.Stat("/run/secrets/redhat.repo")
+	if err != nil {
+		// since the presence of /run/secrets/redhat.repo is not a given, we won't log this a V(0) like
+		// we do for /run/secrets/rhsm
+		log.V(4).Infof("Red Hat entitlements will not be available in this build: failed to stat redhat.repo secrets file /run/secrets/redhat.repo: %v", err)
+		return mounts
+	}
+	if st.IsDir() {
+		// if the file is there, but an unexpected type, then always have log show up via V(0)
+		log.V(0).Infof("Red Hat entitlements will not be available in this build: redhat.repo secrets location /run/secrets/redhat.repo is a directory")
```

Comment:
If my understanding of `redhat.repo` is correct, you can still access subscription content via the CDN without this file. It is only if the subscription content is accessed via satellite will there be an issue.

Therefore I would change the log wording from "will not be available" to "may not be available"

```suggestion
		log.V(4).Infof("Falling back to the Red Hat yum repository configuration in the build's base image: redhat.repo secrets location /run/secrets/redhat.repo is a directory.")
```